### PR TITLE
unittests: Fixes hostutil.GetFileType for Windows

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/sig windows
/sig testing

/kind bug
/kind failing-test
/priority important-soon

#### What this PR does / why we need it:

``GetFileType`` is meant to return the type of the given file by using ``os.Stat``. However, ``os.Stat`` doesn't work on Windows for Unix Sockets, causing an error to occur:

```
[2-Socket Test] unexpected error : CreateFile C:\Users\Administrator\AppData\Local\Temp\test-get-filetype-2776877299\mt.sock: The file cannot be accessed by the system.
```
This is a known issue and we're already using a workaround for this in ``pkg/kubelet/util/util_windows.go``.

This PR fixes this issue for ``GetFileType`` on Windows.
Moves ``IsUnixDomainSocket`` and related unit tests to ``pkg/volume/util/hostutil/hostutil_windows.go``.

#### Which issue(s) this PR fixes:

Related: https://github.com/kubernetes/kubernetes/issues/51540

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by bart0sh** [comment]
@claudiubelu Thank your for your PR!

Generally it's a valid fix. Considering that this code is used in kubelet and it's not directly related to storage, should we try to find a better place for it? I'm not sure it logically belongs to `pkg/volume/util`.

**Comment by SergeyKanzhelev** [comment]
> @claudiubelu Thank your for your PR!
> 
> Generally it's a valid fix. Considering that this code is used in kubelet and it's not directly related to storage, should we try to find a better place for it? I'm not sure it logically belongs to `pkg/volume/util`.

@claudiubelu can you please comment on this?

**Comment by claudiubelu** [comment]
> > @claudiubelu Thank your for your PR!
> > Generally it's a valid fix. Considering that this code is used in kubelet and it's not directly related to storage, should we try to find a better place for it? I'm not sure it logically belongs to `pkg/volume/util`.
> 
> @claudiubelu can you please comment on this?

Moved ``IsUnixDomainSocket`` to ``pkg/util/filesystem``.

**Comment by claudiubelu** [comment]
Note: The Windows unit tests ran on this PR [1], and the following tests that are enabled / touched in this PR passed:

```
k8s.io/kubernetes/pkg/kubelet/util: TestNonUnixDomainSocket
k8s.io/kubernetes/pkg/util/filesystem: TestUnixDomainSocket
k8s.io/kubernetes/pkg/util/filesystem: TestPendingUnixDomainSocket
k8s.io/kubernetes/pkg/volume/util/hostutil: TestGetFileType
```

[1] https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/114336/pull-ci-kubernetes-unit-windows/1618919967732797440

**Review by bart0sh — APPROVED** [review]
/lgtm

**Comment by jsafrane** [comment]
/approve
from sig-storage POV.

**Review by mrunalp — APPROVED** [review]

**Comment by zelenushechka** [comment]
@claudiubelu @SergeyKanzhelev @jsafrane 
Hello! I would like to reach out and confirm if this issue is still on track for the 1.28 release. Could you please provide an update on its progress? Thank you!

**Comment by jsafrane** [comment]
I can only
/lgtm

You need approval of someone from https://github.com/kubernetes/kubernetes/blob/master/pkg/OWNERS to move this further.

**Review by endocrimes — APPROVED** [review]
/lgtm

**Comment by zelenushechka** [comment]
Hi @claudiubelu! I wanted to bring to your attention that the CI is currently failing on your PR. Could you please take a look at it and address the issue? It seems necessary to fix it. Thank you!

**Comment by zelenushechka** [comment]
Week 9 update: Code Freeze approaching!

Hey! We are currently in Week 9, and Code Freeze is just over a week away. It will begin at 01:00 UTC on Wednesday, 19th July 2023 / 18:00 PDT on Tuesday, 18th July 2023. Please ensure all necessary submissions are made before the deadline. Let me know if you need any assistance. :)

**Comment by claudiubelu** [comment]
> Week 9 update: Code Freeze approaching!
> 
> Hey! We are currently in Week 9, and Code Freeze is just over a week away. It will begin at 01:00 UTC on Wednesday, 19th July 2023 / 18:00 PDT on Tuesday, 18th July 2023. Please ensure all necessary submissions are made before the deadline. Let me know if you need any assistance. :)


Simply waiting for reviews / approval at this point.

**Comment by furkatgofurov7** [comment]
@claudiubelu hi, Bug Triage Lead here 👋🏼

Since it was communicated in earlier weeks, both Code / Test Freeze Deadline passed and, this PR has not made to the current milestone, hence I am moving this PR out of the 1.28 milestone

/milestone next-candidate

File: pkg/volume/util/hostutil/hostutil_windows.go
Comment (SergeyKanzhelev) [inline pkg/volume/util/hostutil/hostutil_windows.go:109]: should we only do this on Windows?

Comment (dchen1107) [reply pkg/volume/util/hostutil/hostutil_windows.go:109]: By reading the code here, I think the code path here is only invoked on Windows. Did I read it wrongly?

**Pull request merged**
